File: 박주성/주식가격.py
prices = [1, 2, 3, 2, 3]


def solution(prices):
    answer = []
    le = len(prices)
    result= 0
    for i in range(le):
        result += 1
        count = 0
        for j in range(result,le):
            count += 1 
            if prices[i] > prices[j]:
                break
        answer.append(count)
        

    return answer

File: 유규빈/주식가격.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(size(prices));
    vector<int> stack;

    for (int i=0; i < size(answer); i++){
        answer[i] = size(answer) - i -1;
    }

    for(int i=0; i < size(prices); i++){
        if(stack.empty() || prices[stack.back()] < prices[i]){
            stack.push_back(i);
        }
        else{
            while (!stack.empty() && prices[stack.back()] > prices[i]){
                answer[stack.back()] = i - stack.back();
                stack.pop_back();
            }
            stack.push_back(i);
        }
    }
    return answer;
}

int main() {
    vector<int> test { 1, 2, 3, 2, 3, 1};
    for (auto i : solution(test)){
        cout << i << " ";
    }
    cout << endl;
    return 0;
}

/*

*/